From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Matthias Schiffer <matthias.schiffer@ew.tq-group.com>
Date: Mon, 3 Jun 2024 15:16:45 +0200
Subject: [PATCH] mfd: tqmx86: update TQMX86_IOSIZE_GPIO

With the extension to up to 16 GPIOs, the GPIO controller now uses a
range of 14 registers.

The old definition was actually missing one register (as the interrupt
status register is at base+4), which remained unnoticed because the size
of ioport mappings is not actually used.

Signed-off-by: Matthias Schiffer <matthias.schiffer@ew.tq-group.com>
Upstream-Status: Pending
---
 drivers/mfd/tqmx86.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mfd/tqmx86.c b/drivers/mfd/tqmx86.c
index a4e3ecfccb8ab107249e062f331f537f605844a0..f404f2cbf00623fe6711fe57eb7e3a9edee1daf9 100644
--- a/drivers/mfd/tqmx86.c
+++ b/drivers/mfd/tqmx86.c
@@ -25,7 +25,7 @@
 #define TQMX86_IOBASE_WATCHDOG	0x18b
 #define TQMX86_IOSIZE_WATCHDOG	0x2
 #define TQMX86_IOBASE_GPIO	0x18d
-#define TQMX86_IOSIZE_GPIO	0x4
+#define TQMX86_IOSIZE_GPIO	0xe
 
 #define TQMX86_REG_BOARD_ID	0x00
 #define TQMX86_REG_BOARD_ID_E38M	1
